packages/react-codemirror/src/CypherEditor.tsx

```diff
@@ -161,9 +161,19 @@ export interface CypherEditorProps {
   readonly?: boolean;
 
   /**
-   * String value to assign to the aria-label attribute of the editor
+   * String value to assign to the aria-label attribute of the editor.
    */
   ariaLabel?: string;
+
+  /**
+   * Whether to enable or disable tab focus mode.
+   *
+   * true will allow the brower's default tabbing behavior.
+   * false will enable tab-trapping within the editor.
+   *
+   * @default false
+   */
+  setTabFocusMode?: boolean;
```

Comment:
The default behavior of the codemirror editor is for tab to move focus, I think it'd make sense to conditionally remove the override:

```
    {
      key: 'Tab',
      preventDefault: true,
      run: insertTab,
    },
    {
      key: 'Shift-Tab',
      preventDefault: true,
      run: indentLess,
    },
 ```
 
 Instead of overriding the override. 
 
 Imo it's fine if it's not a reactive setting (i don't see when you'd dynamically toggle this on and off), so the boolean prop can just be passed to the `basicNeo4jSetup`. call in the `componentDidMount`. Not sure if the current solution also does it, but this way tab would still accept completions for example

Reply:
If we want don't want to increase the prop surface, we could also call `this.editorView.current.setTabFocusMode` from outside as it's reachable outside the component through a ref i think

Reply:
I think a `moveFocusOnTab` (or simliarly named) defaulting to false prop could make senes

Reply:
makes total sense

based on our discussion I've renamed the property we're exposing here, removed the setTabFocusMode setting and instead have used the new property to conditionally set the tab keybindings in the setup module

from testing, I found that the tab to accept completions still trapped the focus inside the editor - whilst it was possible to shift-tab _out_ of the editor, tabbing took you straight back in.. so I've included this in the conditional